### DISCUSSION TITLE
Song Select filter should not debounce when search criteria is empty

### DISCUSSION THREAD

**Discussion by Stedoss** [ppy/osu#26625] [Ideas]
In stable, it seems that searches in song select are debounced _unless_ the search is empty (there is no search criteria, eg. a user has pressed `Esc` in song select to clear an existing search). This makes it easy to quickly clear a search, and perform an action (eg. Random/F2) right afterwards without waiting for the filter to be applied the carousel.

In lazer, I have noticed that I have been pressing `Esc` to clear a search and quickly pressing `F2`, but instead of getting a random song from my unfiltered collection I will get a random beatmap from my existing search (the "empty" filter would have not been applied to the carousel due to the pending debounce).

I think it would be nice to avoid this debounce if the search criteria is empty for lazer too.

Locally, this patch feels a lot better, but I'm sure there's a better place to stick the check:
```diff
diff --git a/osu.Game/Screens/Select/SongSelect.cs b/osu.Game/Screens/Select/SongSelect.cs
index bf1724995a..91d83dd2a4 100644
--- a/osu.Game/Screens/Select/SongSelect.cs
+++ b/osu.Game/Screens/Select/SongSelect.cs
@@ -397,7 +397,7 @@ protected override bool OnScroll(ScrollEvent e)
         protected virtual void ApplyFilterToCarousel(FilterCriteria criteria)
         {
             // if not the current screen, we want to get carousel in a good presentation state before displaying (resume or enter).
-            bool shouldDebounce = this.IsCurrentScreen();
+            bool shouldDebounce = this.IsCurrentScreen() && criteria.SearchText.Length > 0;
 
             Carousel.Filter(criteria, shouldDebounce);
         }
```

Video of no-debounce applied:

https://github.com/ppy/osu/assets/29103029/a722e20c-df3b-4add-8fcb-ea5a4155ac09

Sorry if this is not very well written, let me know if there's any more detail I can include!

**Comment by peppy** [comment]
Probably fine to apply this for now. I think the debounce will change more drastically in the future. For instance, changing sort modes also doesn't need debounce (if we assume the debounce is just to avoid filtering from happening while typing).
